src/mongo/db/catalog/collection_options.cpp

```diff
@@ -142,6 +142,8 @@ Status CollectionOptions::parse(const BSONObj& options) {
             cappedSize = e.numberLong();
             if (cappedSize < 0)
                 return Status(ErrorCodes::BadValue, "size has to be >= 0");
+            if ( cappedSize >= 0x7fffffffffffffffLL)
```

Comment:
As there are no `long long` values larger than `0x7fffffffffffffffLL`, this essentially just checks for one specific value, however...